crates/e2e/tests/e2e/cow_amm.rs

```diff
@@ -595,3 +596,685 @@ async fn cow_amm_driver_support(web3: Web3) {
+            (onchain.contracts().weth.address(), to_wei(2000)), // 1 WETH = $2000
+        ]),
+        trades: vec![
+            solvers_dto::solution::Trade::Jit(solvers_dto::solution::JitTrade {
+                order: solvers_dto::solution::JitOrder {
+                    sell_token: cow_amm_order.sell_token,
+                    buy_token: cow_amm_order.buy_token,
+                    receiver: cow_amm_order.receiver.unwrap_or_default(),
+                    sell_amount: cow_amm_order.sell_amount,
+                    buy_amount: cow_amm_order.buy_amount,
+                    valid_to: cow_amm_order.valid_to,
+                    app_data: cow_amm_order.app_data.0,
+                    kind: Kind::Sell,
+                    sell_token_balance: SellTokenBalance::Erc20,
+                    buy_token_balance: BuyTokenBalance::Erc20,
+                    signing_scheme: SigningScheme::Eip1271,
+                    signature,
+                },
+                executed_amount: cow_amm_order.sell_amount - fee_cow_amm,
+                fee: Some(fee_cow_amm),
+            }),
+            solvers_dto::solution::Trade::Fulfillment(solvers_dto::solution::Fulfillment {
+                order: user_order_id.0,
+                executed_amount: user_order.sell_amount - fee_user,
+                fee: Some(fee_user),
+            }),
+        ],
+        pre_interactions: vec![cow_amm_commitment],
+        interactions: vec![],
+        post_interactions: vec![],
+        gas: None,
+    };
+    mock_solver.configure_solution(Some(mocked_solution.clone()));
+
+    // Drive solution
+    tracing::info!("Waiting for trade.");
+    wait_for_condition(TIMEOUT, || async {
+        let amm_weth_balance_after = onchain
+            .contracts()
+            .weth
+            .balance_of(cow_amm.address())
+            .call()
+            .await
+            .unwrap();
+        let bob_weth_balance_after = onchain
+            .contracts()
+            .weth
+            .balance_of(bob.address())
+            .call()
+            .await
+            .unwrap();
+
+        let amm_weth_sent = amm_weth_balance_before - amm_weth_balance_after;
+        let bob_weth_received = bob_weth_balance_after - bob_weth_balance_before;
+
+        // Bob should receive WETH, CoW AMM's WETH balance decreases
+        bob_weth_received >= user_order.buy_amount && amm_weth_sent >= cow_amm_order.sell_amount
+    })
+    .await
+    .unwrap();
+
+    // Verify that the trade is indexed
+    tracing::info!("Waiting for trade to be indexed.");
+    wait_for_condition(TIMEOUT, || async {
+        let trades = services.get_trades(&user_order_id).await.ok();
+        trades.is_some_and(|trades| !trades.is_empty())
+    })
+    .await
+    .unwrap();
+}
+
+#[tokio::test]
+#[ignore]
+async fn local_node_cow_amm_quoting() {
+    run_test(cow_amm_quoting).await;
+}
+
+/// Tests quotes verification that contains CoW AMM JIT orders.
+async fn cow_amm_quoting(web3: Web3) {
```

Comment:
I see there is a separate test for quoting specifically now. Can this not be merged with the other new test?

Reply:
https://github.com/cowprotocol/services/pull/3038#discussion_r1796780130